users/serializers.py

```diff
@@ -1 +1,84 @@
-from django.contrib.auth import get_user_model
from django.core.exceptions import ValidationError
from djoser.serializers import UserCreateSerializer


User = get_user_model()


class CustomUserCreateSerializer(UserCreateSerializer):
    class Meta(UserCreateSerializer.Meta):
        model = User
        fields = ('email', 'first_name', 'last_name', 'user_phone',
                  'password', 'about_me', 'profile_picture', 'roles')

    def create(self, validated_data):
        roles = validated_data.pop('roles', [])
        user = super().create(validated_data)
        for role in roles:
            try:
                user.add_role(role.name)
            except ValidationError as e:
                raise ValidationError(f"Failed to add role '{role.name}': {e}")
        return user
\ No newline at end of file
+from django.contrib.auth import get_user_model
+from django.contrib.auth.password_validation import validate_password
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+from .models import Role
+
+
+User = get_user_model()
+
+
+class CustomUserCreateSerializer(serializers.ModelSerializer):
+    """
+    Custom serializer for creating and updating users.
+
+    Attributes:
+        roles (PrimaryKeyRelatedField): Field for associating roles with a user.
+
+    Meta:
+        model (User): The custom user model.
+        fields (tuple): List of fields to include in the serialization.
+        extra_kwargs (dict): Additional options for some fields (e.g. password write-only).
+    """
+
+    roles = serializers.PrimaryKeyRelatedField(queryset=Role.objects.all(), many=True, required=False)
+
+    class Meta:
+        model = User
+        fields = ('email', 'first_name', 'last_name', 'user_phone',
+                  'password', 'about_me', 'profile_picture', 'roles')
+        extra_kwargs = {
+            'password': {'write_only': True},
+            'profile_picture': {'required': False, 'allow_null': True},
+        }
+
+    def create(self, validated_data):
+        """
+        Create a new user instance with validated data.
+
+        Args:
+            validated_data (dict): The validated data for creating the user.
+
+        Returns:
+            User: The created user instance.
+        """
+        roles_data = validated_data.pop('roles', [])
+        user = User.objects.create(**validated_data)
+        user.save()
+        if roles_data:
+            user.roles.set(roles_data)
+        return user
```

Comment:
When creating or updating a user with multiple related objects (like roles), it's safer to use a transaction to ensure all operations succeed or are rolled back if one fails. This prevents cases where the user might be created without proper role assignments due to an error.
```
from django.db import transaction

def create(self, validated_data):
    roles_data = validated_data.pop('roles', [])
    with transaction.atomic():
        user = User.objects.create(**validated_data)
        user.set_password(validated_data['password'])
        user.save()
        if roles_data:
            user.roles.set(roles_data)
    return user
```